src/main/scala/sample/reactivekafka/KafkaReaderCoordinator.scala

```diff
@@ -41,19 +41,46 @@ class KafkaReaderCoordinator(mat: Materializer, topicName: String) extends Actor
       .kafkaOffsetsStorage()
       .commitInterval(1200 milliseconds)
     )
+
     log.debug("Starting the reader")
-    Source(consumerWithOffsetSink.publisher)
-      .map(processMessage)
-      .withAttributes(ActorAttributes.supervisionStrategy(processingDecider))
-      .to(consumerWithOffsetSink.offsetCommitSink).run()
```

Comment:
Yes, because theoretically 'processMessage' can fail. In this example it will obviously always pass but in real life scenarios something bad may happen. Maybe a better example would be something that randomly fails in the `processMessage` function.